contracts/src/v0.8/shared/enumerable/CCIPEnumerableMap.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {EnumerableSet} from "../../vendor/openzeppelin-solidity/v4.8.3/contracts/utils/structs/EnumerableSet.sol";
+
+/**
+ * @dev Library for managing an enumerable variant of Solidity's
+ * https://solidity.readthedocs.io/en/latest/types.html#mapping-types[`mapping`]
+ * type.
+ *
+ * Maps have the following properties:
+ *
+ * - Entries are added, removed, and checked for existence in constant time
+ * (O(1)).
+ * - Entries are enumerated in O(n). No guarantees are made on the ordering.
+ *
+ * ```
+ * contract Example {
+ *     // Add the library methods
+ *     using CCIPEnumerableMap for CCIPEnumerableMap.Bytes32ToBytesMap;
+ *
+ *     // Declare a set state variable
+ *     CCIPEnumerableMap.Bytes32ToBytesMap private myMap;
+ * }
+ * ```
+ *
+ * The following map types are supported:
+ *
+ * - `bytes32 -> bytes` (`Bytes32ToBytes`)
+ *
+ * [WARNING]
+ * ====
+ * Trying to delete such a structure from storage will likely result in data corruption, rendering the structure
+ * unusable.
+ * See https://github.com/ethereum/solidity/pull/11843[ethereum/solidity#11843] for more info.
+ *
+ * In order to clean an CCIPEnumerableMap, you can either remove all elements one by one or create a fresh instance using an
+ * array of CCIPEnumerableMap.
+ * ====
+ */
+library CCIPEnumerableMap {
```

Comment:
let's remove the CCIP naming

Reply:
+1, something like `EnumerableMapBytes32` to be consistent with `EnumerableMapAddresses`